From ab2a5fd6b3b2384fa7ec975c1b1b4754eb540d47 Mon Sep 17 00:00:00 2001
From: Hannes Winkler <hanneswinkler2000@web.de>
Date: Thu, 22 Jun 2023 16:30:00 +0000
Subject: [PATCH 3/3] always include debug symbols

Always build with `-ggdb -fdebug-default-version=4` by default.
Symbols are stripped later anyway and they're still useful for debugging.
---
 build/config/compiler/BUILD.gn | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/build/config/compiler/BUILD.gn b/build/config/compiler/BUILD.gn
index cd2e662..cce2c1a 100644
--- a/build/config/compiler/BUILD.gn
+++ b/build/config/compiler/BUILD.gn
@@ -1015,7 +1015,7 @@ config("symbols") {
       ldflags = [ "-gsource-map"]
     }
   } else {
-    cflags = [ "-g2" ]
+    cflags = [ "-ggdb", "-fdebug-default-version=4" ]
   }
 }
 
@@ -1024,12 +1024,12 @@ config("minimal_symbols") {
     # Linker symbols for backtraces only.
     ldflags = [ "/DEBUG" ]
   } else {
-    cflags = [ "-g1" ]
+    cflags = [ "-ggdb", "-fdebug-default-version=4" ]
   }
 }
 
 config("no_symbols") {
   if (!is_win) {
-    cflags = [ "-g0" ]
+    cflags = [ "-ggdb", "-fdebug-default-version=4" ]
   }
 }
-- 
2.39.2

